### DISCUSSION TITLE
Linkerd Check as an API Call?

### DISCUSSION THREAD

**Discussion by manju-rn** [linkerd/linkerd2#11574] [General]
The `linkerd check` is very useful, especially `linkerd mc check` for the multicluster setup is very useful for me. However, running as a CLI restricts the automation option. Is there an API (preferably rest based) which can be called to do this exact checks and returns the results in json just like `linkerd mc check -o json` returns?
If there are client libs (in whatever language) doing the same also will be a good alternative.
I plan to run this as a k8s cron-job

**Comment by alpeb** [comment]
Closing, as already posted in #11575 and [here](https://linkerd.buoyant.io/t/linkerd-check-as-an-api-call/312).
